spec/decorator_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative './../decorator'
+require_relative './../person'
+
+describe Decorator do
+  context 'When testing the Decorator class' do
+    it 'should create a new decorator when Decorator class is instantiated' do
+      p = Person.new(34, 'Duane')
+      decorator = Decorator.new(p)
+      expect(decorator.correct_name).to eq 'Duane'
+    end
+  end
+
+  context 'When testing the Decorator class' do
+    it 'should create a new decorator when Decorator class is instantiated' do
+      p = Person.new(34, 'DuaneStevenson')
+      decorator = TrimmerDecorator.new(p)
+      expect(decorator.correct_name).to eq 'DuaneSteve'
+    end
+  end
+
+  context 'When testing the Decorator class' do
+    it 'should create a new decorator when Decorator class is instantiated' do
+      p = Person.new(34, 'duane')
```

Comment:
You are missing tests for the class in `person.rb` file. Kindly consider the following guidelines for testing methods as well.

- Test for `add_rental` method
e.g 
```
  context "When adding a new rental" do

    it "The add_rental method returns a rental and adds its to the book" do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Bob")

      rental = person.add_rental(book, "2021-05-04")

      expect(person.rentals).to eql [rental]
    end
  end
```

Checking permission using the `can_use_services?` method
- `can_use_services?` negative scenario: The can_use_services? method returns false if under age and no parent permission"
e.g 
```
    it "The can_use_services? method returns false if under age and no parent permission" do
      person = Person.new(14, "Rose", false)
      expect(person.can_use_services?).to be false
    end
```

- `can_use_services?` positive scenario: The can_use_services? method returns true if over age
- `can_use_services?` positive scenario: The can_use_services? method returns true if has parent permission